cdap-api/src/main/java/co/cask/cdap/api/messaging/MessagingAdmin.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.messaging;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Provides topic administration functions on the transactional messaging system.
+ */
+@Beta
+public interface MessagingAdmin {
+
+  /**
+   * Creates a new topic with default topic properties.
+   *
+   * @param topic name of the topic
+   * @throws IOException if failed to communicate with the messaging system. Caller may retry this call or
+   *                     use the {@link #getTopicProperties(String)} method verify if the topic has been created.
+   * @throws IllegalArgumentException if the topic name is invalid. A valid id should only contain alphanumeric
+   *                                  characters and {@code _} or {@code -}.
+   * @throws IllegalStateException if the topic already exist.
```

Comment:
IllegalStateException is for reporting that the application got into an inconsistent state. A client requesting a non-existent topic is not an illegal state, it is an expected user error. I think it is important that developers who use this API know what exceptions to expect, and checked exceptions are the best way to do that. 

IllegalArgumentException itself is a border case. We throw it in many places - for internal errors as well for client errors. This makes it hard for clients to know whether they passed in illegal arguments, or whether the system has a bug and uses an illegal argument internally. 

For example, Integer.parseInteger throws NumberFormatException for that reason. MalformedURLException is a similar example. Also Java's IO classes throw IOExceptions (actually FileNotFoundException) if a file does not exist.  

How about simply introducing a TopicNotFoundException?